okteto/okteto
# Added OKTETO_GIT_BRANCH environment variable to remote Dockerfile as argument to have it available in the commands executed there

# Proposed changes

As part of the remote execution of deploy and destroy commands, we are passing the environment variable `OKTETO_GIT_COMMIT` to the remote execution, but we were not passing the environment variable `OKTETO_GIT_BRANCH`. That is a variable we have available locally and in the installer when deploying dev environments and some users use it in their flows.

This PR is adding `OKTETO_GIT_BRANCH` to the remote execution of deploy and destroy to make it available that env var like we do with the commit

## How to validate

Steps to reproduce the issue:

1. In a simple repo, create a file `.oktetodeployignore` to not sync `.git/` folder when executing deploy remotely
2. Define a simple manifest like this one:

```yaml
deploy:
  commands:
    - name: "echo git commit"
      command: echo $OKTETO_GIT_COMMIT
    - name: "echo git branch"
      command: echo $OKTETO_GIT_BRANCH
destroy:
  commands:
    - name: "echo git commit"
      command: echo $OKTETO_GIT_COMMIT
    - name: "echo git branch"
      command: echo $OKTETO_GIT_BRANCH
```

3. Execute `okteto deploy -f <path-to-manifest> --remote`
4. You will see that the second command won't display anything

If you follow the same steps with the code in this branch, it will print the branch correctly.

I was testing okteto deploy local, remote and from the installer and everything is working fine

## CLI Quality Reminders 🔧

For both authors and reviewers:

- Scrutinize for potential regressions
- Ensure key automated tests are in place
- Build the CLI and test using the validation steps
- Assess Developer Experience impact (log messages, performances, etc)
- If too broad, consider breaking into smaller PRs
- Adhere to our [code style](https://github.com/okteto/okteto/blob/master/docs/code-style.md) and [code review](https://github.com/okteto/okteto/blob/master/docs/code-review.md) guidelines
